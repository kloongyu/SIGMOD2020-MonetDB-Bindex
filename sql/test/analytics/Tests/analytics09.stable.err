stderr of test 'analytics09` in directory 'sql/test/analytics` itself:


# 13:13:17 >  
# 13:13:17 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=37694" "--set" "mapi_usock=/var/tmp/mtest-1389/.s.monetdb.37694" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/ferreira/repositories/MonetDB-analytics/BUILD/var/MonetDB/mTests_sql_test_analytics" "--set" "embedded_c=true"
# 13:13:17 >  

# builtin opt 	gdk_dbpath = /home/ferreira/repositories/MonetDB-analytics/BUILD/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 37694
# cmdline opt 	mapi_usock = /var/tmp/mtest-1389/.s.monetdb.37694
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /home/ferreira/repositories/MonetDB-analytics/BUILD/var/MonetDB/mTests_sql_test_analytics
# cmdline opt 	embedded_c = true
# cmdline opt 	gdk_debug = 553648138

# 13:13:17 >  
# 13:13:17 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-1389" "--port=37694"
# 13:13:17 >  

MAPI  = (monetdb) /var/tmp/mtest-17924/.s.monetdb.36475
QUERY = select cast(sum(aa) * 100 / sum(sum(aa)) over (partition by bb) as bigint) from analytics; --error, nesting aggregation functions
ERROR = !SELECT: identifier 'bb' unknown
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-1389/.s.monetdb.37694
QUERY = select cast(prod(sum(aa)) * count(1 + aa) / avg(null) over () as bigint) from analytics; --error, nesting aggregation functions
ERROR = !SELECT: subquery result missing
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-17541/.s.monetdb.31437
QUERY = select rank() over (partition by sum(aa), bb) from analytics group by aa; --error
ERROR = !SELECT: identifier 'bb' unknown
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-10156/.s.monetdb.31763
QUERY = select min(aa) over (partition by sum(bb)) from analytics; --error
ERROR = !SELECT: cannot use non GROUP BY column 'aa' in query results without an aggregate function
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-9405/.s.monetdb.32071
QUERY = select min(aa) over (partition by sum(aa)) from analytics; ---error
ERROR = !SELECT: cannot use non GROUP BY column 'aa' in query results without an aggregate function
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-9623/.s.monetdb.34265
QUERY = select rank() over (order by sum(aa), bb) from analytics; --error
ERROR = !SELECT: cannot use non GROUP BY column 'bb' in query results without an aggregate function
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-9623/.s.monetdb.34265
QUERY = select min(aa) over (order by sum(bb)) from analytics; --error
ERROR = !SELECT: cannot use non GROUP BY column 'aa' in query results without an aggregate function
CODE  = 42000

# 13:13:17 >  
# 13:13:17 >  "Done."
# 13:13:17 >  

