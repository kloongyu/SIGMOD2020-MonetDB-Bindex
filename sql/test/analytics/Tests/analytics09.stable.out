stdout of test 'analytics09` in directory 'sql/test/analytics` itself:


# 13:13:17 >  
# 13:13:17 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=37694" "--set" "mapi_usock=/var/tmp/mtest-1389/.s.monetdb.37694" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/ferreira/repositories/MonetDB-analytics/BUILD/var/MonetDB/mTests_sql_test_analytics" "--set" "embedded_c=true"
# 13:13:17 >  

# MonetDB 5 server v11.32.0
# This is an unreleased version
# Serving database 'mTests_sql_test_analytics', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit
# Found 15.492 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2018 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://wired-142.cwi.nl:37694/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-1389/.s.monetdb.37694
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

Ready.

# 13:13:17 >  
# 13:13:17 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-1389" "--port=37694"
# 13:13:17 >  

#create table analytics (aa int, bb int);
#insert into analytics values (15, 3), (3, 1), (2, 1), (5, 3), (NULL, 2), (3, 2), (4, 1), (6, 3), (8, 2), (NULL, 4);
[ 10	]
#select cast(sum(1) over () as bigint), rank() over (), nth_value(1, 1) over ();
% .L5,	.L10,	.L14 # table_name
% L5,	L10,	L14 # name
% bigint,	int,	tinyint # type
% 1,	1,	1 # length
[ 1,	1,	1	]
#select avg(sum(aa) over ()) from analytics;
% sys.L10 # table_name
% L10 # name
% double # type
% 24 # length
[ 46	]
#select cast(sum(1) * count(*) over () as bigint);
% .L7 # table_name
% L7 # name
% bigint # type
% 1 # length
[ 1	]
#select cast(sum(aa) * count(*) over () as bigint) from analytics;
% sys.L7 # table_name
% L7 # name
% bigint # type
% 2 # length
[ 46	]
#select cast(aa * count(1) over () as bigint) from analytics;
% sys.L5 # table_name
% L5 # name
% bigint # type
% 3 # length
[ 150	]
[ 30	]
[ 20	]
[ 50	]
[ NULL	]
[ 30	]
[ 40	]
[ 60	]
[ 80	]
[ NULL	]
#select cast(sum(aa) * count(1) over () as bigint) from analytics;
% sys.L7 # table_name
% L7 # name
% bigint # type
% 2 # length
[ 46	]
#select cast(sum(aa) * count(1 + aa) / avg(1) over () as bigint) from analytics;
% sys.L10 # table_name
% L10 # name
% bigint # type
% 3 # length
[ 368	]
#select avg(sum(aa)) over () from analytics;
% sys.L5 # table_name
% L5 # name
% double # type
% 24 # length
[ 46	]
#select sum(cast(aa as double)) over (rows unbounded preceding) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 15	]
[ 18	]
[ 20	]
[ 25	]
[ 25	]
[ 28	]
[ 32	]
[ 38	]
[ 46	]
[ 46	]
#select sum(cast(aa as double)) over (range unbounded preceding) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 46	]
[ 46	]
[ 46	]
[ 46	]
[ 46	]
[ 46	]
[ 46	]
[ 46	]
[ 46	]
[ 46	]
#select avg(avg(aa)) over (rows unbounded preceding) from analytics;
% sys.L5 # table_name
% L5 # name
% double # type
% 24 # length
[ 5.75	]
#select avg(avg(aa)) over (range unbounded preceding) from analytics;
% sys.L5 # table_name
% L5 # name
% double # type
% 24 # length
[ 5.75	]
#select avg(sum(aa)) over (rows unbounded preceding) from analytics;
% sys.L5 # table_name
% L5 # name
% double # type
% 24 # length
[ 46	]
#select avg(sum(aa)) over (range unbounded preceding) from analytics;
% sys.L5 # table_name
% L5 # name
% double # type
% 24 # length
[ 46	]
#select avg(sum(aa)) over (), avg(avg(aa)) over () from analytics;
% sys.L5,	sys.L11 # table_name
% L5,	L11 # name
% double,	double # type
% 24,	24 # length
[ 46,	5.75	]
#select avg(sum(aa)) over (),
#       cast(sum(aa) * count(case when bb < 2 then bb - 1 else bb + 1 end) / avg(1) over (rows between current row and current row) as bigint),
#       avg(sum(aa)) over (rows unbounded preceding),
#       avg(sum(aa)) over (range unbounded preceding) from analytics;
% sys.L5,	sys.L16,	sys.L21,	sys.L25 # table_name
% L5,	L16,	L21,	L25 # name
% double,	bigint,	double,	double # type
% 24,	3,	24,	24 # length
[ 46,	460,	46,	46	]
#select avg(sum(aa)) over () from analytics group by aa;
% sys.L3 # table_name
% L3 # name
% double # type
% 24 # length
[ 6.571428571	]
[ 6.571428571	]
[ 6.571428571	]
[ 6.571428571	]
[ 6.571428571	]
[ 6.571428571	]
[ 6.571428571	]
[ 6.571428571	]
#select cast(sum(aa) * count(aa) / avg(aa) over (rows between current row and unbounded following) as bigint) from analytics group by aa;
% sys.L6 # table_name
% L6 # name
% bigint # type
% 1 # length
[ 2	]
[ 3	]
[ 0	]
[ 1	]
[ NULL	]
[ 1	]
[ 1	]
[ 1	]
#select avg(sum(aa)) over (),
#       avg(sum(aa)) over (rows unbounded preceding),
#       cast(sum(aa) * count(aa) / avg(aa) over (rows between current row and unbounded following) as bigint),
#       avg(sum(aa)) over (range unbounded preceding) from analytics group by aa;
% sys.L3,	sys.L7,	sys.L16,	sys.L21 # table_name
% L3,	L7,	L16,	L21 # name
% double,	double,	bigint,	double # type
% 24,	24,	1,	24 # length
[ 6.571428571,	15,	2,	6.571428571	]
[ 6.571428571,	10.5,	3,	6.571428571	]
[ 6.571428571,	7.666666667,	0,	6.571428571	]
[ 6.571428571,	7,	1,	6.571428571	]
[ 6.571428571,	7,	NULL,	6.571428571	]
[ 6.571428571,	6.4,	1,	6.571428571	]
[ 6.571428571,	6.333333333,	1,	6.571428571	]
[ 6.571428571,	6.571428571,	1,	6.571428571	]
#select cast(sum(aa) * count(aa) over () as bigint),
#       cast(sum(aa) over () as bigint) from analytics group by aa;
% sys.L5,	sys.L11 # table_name
% L5,	L11 # name
% bigint,	bigint # type
% 3,	2 # length
[ 105,	43	]
[ 42,	43	]
[ 14,	43	]
[ 35,	43	]
[ NULL,	43	]
[ 28,	43	]
[ 42,	43	]
[ 56,	43	]
#select cast(sum(sum(aa)) over () as bigint),
#       cast(sum(aa) * count(count(aa)) over () as bigint) from analytics group by aa;
% sys.L5,	sys.L13 # table_name
% L5,	L13 # name
% bigint,	bigint # type
% 2,	3 # length
[ 46,	120	]
[ 46,	48	]
[ 46,	16	]
[ 46,	40	]
[ 46,	NULL	]
[ 46,	32	]
[ 46,	48	]
[ 46,	64	]
#select count(aa) over (),
#       avg(aa) over () * count(aa) from analytics group by aa;
% sys.L2,	sys.L10 # table_name
% L2,	L10 # name
% bigint,	double # type
% 1,	24 # length
[ 7,	6.142857143	]
[ 7,	12.28571429	]
[ 7,	6.142857143	]
[ 7,	6.142857143	]
[ 7,	0	]
[ 7,	6.142857143	]
[ 7,	6.142857143	]
[ 7,	6.142857143	]
#select cast(sum(aa) over () as bigint),
#       cast(sum(aa) over () as bigint),
#       cast(sum(aa) * count(aa) over () as bigint) from analytics group by aa;
% sys.L4,	sys.L10,	sys.L15 # table_name
% L4,	L10,	L15 # name
% bigint,	bigint,	bigint # type
% 2,	2,	3 # length
[ 43,	43,	105	]
[ 43,	43,	42	]
[ 43,	43,	14	]
[ 43,	43,	35	]
[ 43,	43,	NULL	]
[ 43,	43,	28	]
[ 43,	43,	42	]
[ 43,	43,	56	]
#select 21 - avg(sum(aa)) over (),
#       avg(45 * count(aa) + sum(aa)) over (),
#       cast(sum(aa) * count(aa) over () as bigint) from analytics group by aa;
% .L5,	sys.L10,	sys.L16 # table_name
% L5,	L10,	L16 # name
% double,	double,	bigint # type
% 24,	24,	3 # length
[ 14.42857143,	58,	105	]
[ 14.42857143,	58,	42	]
[ 14.42857143,	58,	14	]
[ 14.42857143,	58,	35	]
[ 14.42857143,	58,	NULL	]
[ 14.42857143,	58,	28	]
[ 14.42857143,	58,	42	]
[ 14.42857143,	58,	56	]
#select avg(sum(aa)) over (partition by bb) from analytics group by bb;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 9	]
[ 11	]
[ 26	]
[ NULL	]
#select cast(sum(aa) * 100 / sum(sum(aa)) over () as bigint) from analytics;
% sys.L7 # table_name
% L7 # name
% bigint # type
% 3 # length
[ 100	]
#select cast(sum(aa) * 100 / sum(sum(aa)) over (partition by bb) as bigint) from analytics group by bb;
% sys.L7 # table_name
% L7 # name
% bigint # type
% 3 # length
[ 100	]
[ 100	]
[ 100	]
[ NULL	]
#select rank() over (partition by case when aa > 5 then aa else aa + 5 end) from analytics;
% sys.L3 # table_name
% L3 # name
% int # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select rank() over (partition by sum(aa)) from analytics;
% sys.L5 # table_name
% L5 # name
% int # type
% 1 # length
[ 1	]
#select rank() over (partition by 12*sum(aa)) from analytics;
% sys.L6 # table_name
% L6 # name
% int # type
% 1 # length
[ 1	]
#select rank() over (partition by sum(aa)) from analytics group by aa;
% sys.L4 # table_name
% L4 # name
% int # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select rank() over (partition by sum(aa)) from analytics group by bb;
% sys.L4 # table_name
% L4 # name
% int # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select rank() over (partition by sum(aa)*sum(bb)) from analytics;
% sys.L7 # table_name
% L7 # name
% int # type
% 1 # length
[ 1	]
#select rank() over (partition by sum(aa), sum(bb)) from analytics;
% sys.L6 # table_name
% L6 # name
% int # type
% 1 # length
[ 1	]
#select rank() over (partition by sum(aa), sum(bb)) from analytics group by aa;
% sys.L5 # table_name
% L5 # name
% int # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select rank() over (order by sum(aa)) from analytics;
% sys.L6 # table_name
% L6 # name
% int # type
% 1 # length
[ 1	]
#select rank() over (order by sum(aa), sum(bb)) from analytics;
% sys.L7 # table_name
% L7 # name
% int # type
% 1 # length
[ 1	]
#select dense_rank() over (partition by sum(aa) order by avg(bb)) from analytics;
% sys.L6 # table_name
% L6 # name
% int # type
% 1 # length
[ 1	]
#drop table analytics;

# 13:13:17 >  
# 13:13:17 >  "Done."
# 13:13:17 >  

